src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -876,11 +876,28 @@ export default {
             params: { correctedFilingId: item.filingId } })
           break
         case FilingTypes.INCORPORATION_APPLICATION:
-          // redirect to Edit web app to correct this Incorporation Application
-          const editUrl = sessionStorage.getItem('EDIT_URL')
-          const url = `${editUrl}${this.entityIncNo}/correction?corrected-id=${item.filingId}`
-          // assume Correct URL is always reachable
-          window.location.assign(url)
+          try {
+            // Fetch original Incorporation Application
+            const iaFiling = await this.fetchFilingById(this.getIncorporationNumber, item.filingId)
+
+            // Create a Draft Incorporation Application Correction Filing
+            const correctionIaFiling = this.buildIaCorrectionFiling(iaFiling)
+            const draftCorrection = await this.createCorrection(this.getIncorporationNumber, correctionIaFiling)
+
+            // Retrieve the Filing ID from the newly created Draft
+            const draftCorrectionId = draftCorrection.header.filingId
+
+            // redirect to Edit web app to correct this Incorporation Application
+            const editUrl = sessionStorage.getItem('EDIT_URL')
+            const url = `${editUrl}${this.getIncorporationNumber}/correction?correction-id=${draftCorrectionId}`
+
+            // assume Correct URL is always reachable
+            window.location.assign(url)
+          } catch (error) {
+            // eslint-disable-next-line no-console
+            console.log(`Correction Creation error = ${error}`)
+            this.downloadErrorDialog = true
```

Comment:
Maybe not the right dialog.

Reply:
ah thanks, i just implemented this for testing. I'm not exactly sure which Dialog to use here to be honest.

Reply:
I was looking at what we do on other Todo List errors... and the answer is not much.
But here we really should let the user know that we can't create a draft correction. Maybe something like the Save Error Dialog or Resume Error Dialog in other views?

Reply:
K, sounds good, will do!

Reply:
Updated to  `LoadCorrection Dialog` with a small tweak there.